### PR TITLE
Fix statistics GROUP BY standard violations

### PR DESCRIPTION
**Pull request by Artanicus**
Now compatible with ONLY_FULL_GROUP_BY which has been the default and
heavily recommended since MySQL 5.7. Fixed in a way that is also compatible with MariaDB.

+ some capitalization fixes for queries just so I could inspect them easier.

### CODE REVIEW

**Comment by Expl0dingBanana** [comment]
@DrMurx do you have any issues with requiring 'ONLY_FULL_GROUP_BY'?

@Artanicus can you fix the merge conflict?

**Comment by Artanicus** [comment]
Conflict fixed.

To clarify, these changes do not mean ONLY_FULL_GROUP_BY must be enabled, rather that the queries work even if it is enabled.

**Comment by DrMurx** [comment]
> @DrMurx do you have any issues with requiring 'ONLY_FULL_GROUP_BY'?

I'm not really having a stake at that code (just refactored it out of the humongous `DbWrapper`). But I'm absolutely for standard compliance, so doing aggregate queries properly is 👍👍👍.

However, removing `ONLY_FULL_GROUP_BY` from the SQL Mode blocker means that for development and testing, this mode **has** to be enabled to spot flawed queries in the future before they're merged into production code.

**Comment by Artanicus** [comment]
@DrMurx Good point. It could be added as a recommendation for developers but not everyone will enable it since if they run any non-compliant software off the same DB server you can't really. In some ways it's up to code review to catch faulty GROUP BY statements but it's not easy for us dinosaurs used to the undefined behavior of years past. Ultimately it might fall on users to catch when it breaks since many new users will have it enabled.

TL;DR; multi-pronged approach is probably the best:
- Amend dev guide to recommend enabling ONLY_FULL_GROUP_BY to catch undefined behavior
- Write tests for the CI to catch it. (Are there even db tests now? Should probably look into it.)
- Try to catch them in code review where the tests can't.
- New users will surely report broken statistics eventually if all else fails.

**Comment by Expl0dingBanana** [comment]
@Artanicus 
Are you currently running this on your system without issues?

**Comment by Artanicus** [comment]
I'm running this without issues. The only places where GROUP BY is used is in the statistics and once in the DBWrapper. The wrapper use-case is dead simple and was already compliant, I just did cosmetic fixes there. For statistics everything works fine that I am actively using but will send out a Mon scanner now to see if those stats also work.

**Comment by Artanicus** [comment]
Can confirm, all statistics views seem to be working normally. (100% IV and shiny views I can't confirm since just enabled Mon scanning.)

**Comment by JabLuszko** [comment]
Does your "per device" stats: MADmin -> Statistics -> Global stats -> (click device name) works?
It's broken here due to `MIN()` in group by and complains from MySQL: ERROR 1111 (HY000): Invalid use of group function

https://github.com/Map-A-Droid/MAD/commit/42e07f2c5246085035ed7203b3fe4d3f7884804e#diff-157e56d1cbb5da5e4ac3b9ef2f372a04R82
